src/patternfly/components/TreeView/tree-view.scss

```diff
@@ -0,0 +1,196 @@
+  --pf-c-tree-view__search--PaddingLeft: var(--pf-global--spacer--sm);
+
+  // icon
+  --pf-c-tree-view__link-icon--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-tree-view__link-icon--Color: var(--pf-global--icon--Color--light);
+
+  // action
+  --pf-c-tree-view__action--PaddingRight: var(--pf-global--spacer--md);
+  --pf-c-tree-view__action--PaddingLeft: var(--pf-global--spacer--md);
+  --pf-c-tree-view__action--Color: var(--pf-global--icon--Color--light);
+  --pf-c-tree-view__action--m-favorite--Color: var(--pf-global--palette--gold-400);
+
+  padding-top: var(--pf-c-tree-view--PaddingTop);
+  padding-bottom: var(--pf-c-tree-view--PaddingBottom);
+}
+
+// stylelint-disable
+.pf-c-tree-view__list {
+  &.pf-m-nested-level-1 {
+    .pf-c-tree-view__list-item {
+      &.pf-m-expandable > .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-1__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-1__list-item__link--PaddingLeft);
+      }
+    }
+  }
+
+  &.pf-m-nested-level-2 {
+    .pf-c-tree-view__list-item {
+      &.pf-m-expandable > .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-2__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-2__list-item__link--PaddingLeft);
+      }
+    }
+  }
+
+  &.pf-m-nested-level-3 {
+    .pf-c-tree-view__list-item {
+      &.pf-m-expandable > .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item--m-expandable__link--PaddingLeft);
+      }
+
+      .pf-c-tree-view__link {
+        --pf-c-tree-view__link--PaddingLeft: var(--pf-c-tree-view__list--m-nested-level-3__list-item__link--PaddingLeft);
+      }
+    }
+  }
+}
+// stylelint-enable
+
+.pf-c-tree-view__list-item {
+  &.pf-m-expanded {
+    > .pf-c-tree-view__link > .pf-c-tree-view__link-toggle {
+      transform: rotate(var(--pf-c-tree-view__link--m-expanded__toggle--Rotate));
+    }
+  }
+}
+
+.pf-c-tree-view__link {
+  display: flex;
+  align-items: center;
+  width: 100%;
+  padding: var(--pf-c-tree-view__link--PaddingTop) var(--pf-c-tree-view__link--PaddingRight) var(--pf-c-tree-view__link--PaddingBottom) var(--pf-c-tree-view__link--PaddingLeft);
+  color: var(--pf-c-tree-view__link--Color);
+  text-align: left;
+  border: 0;
+
+  &.pf-m-current {
+    --pf-c-tree-view__link--Color: var(--pf-c-tree-view__link--m-current--Color);
+
+    font-weight: var(--pf-c-tree-view__link--m-current--FontWeight);
+  }
+
+  &.pf-m-first {
+    font-weight: var(--pf-c-tree-view__link--m-first--FontWeight);
+  }
+
+  &:hover {
+    background-color: var(--pf-c-tree-view__link--hover--BackgroundColor);
+  }
+
+  .pf-c-check {
```

Comment:
If we're just placing a checkbox there, I think we should just use a normal `<input type="checkbox">`. That or update the check component so it can work without a label - I just created an issue for that here https://github.com/patternfly/patternfly/issues/3311

Reply:
Sounds good, I'll use a normal input for now and we could follow up with adding the check component here once the issue is finished.